src/Uuid/FileUuid.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace Kirby\Uuid;
+
+use Generator;
+use Kirby\Cms\File;
+
+/**
+ * Uuid for \Kirby\Cms\File
+ *
+ * @package   Kirby Uuid
+ * @author    Nico Hoffmann <nico@getkirby.com>
+ * @link      https://getkirby.com
+ * @copyright Bastian Allgeier
+ * @license   https://getkirby.com/license
+ */
+class FileUuid extends ModelUuid
+{
+	protected const TYPE = 'file';
+
+	/**
+	 * @var \Kirby\Cms\File|null
+	 */
+	public Identifiable|null $model;
+
+	/**
+	 * Look up Uuid in cache and resolve to file object.
+	 * Special for  FileUuid as the value stored in cache
+	 * is a hybrid Uri from parent Uuid and filename. Needs
+	 * to resolve parent Uuid and then get file by filename.
+	 */
+	protected function findByCache(): File|null
+	{
+		// get mixed Uri from cache
+		$key   = $this->key();
+		$value = Uuids::cache()->get($key);
+
+		if ($value === null) {
+			return null;
+		}
+
+		// value is itself another Uuid\Uri string
+		// e.g. page://page-uuid/filename.jpg
+		$uuid = new Uri($value);
+
+		// we need to resolve the parent UUID to its model
+		// and then query for the file by filename
+		$parent   = Uuid::for($uuid->base())->resolve();
```

Comment:
I'd be fine with either way - if the first we should document it as well as check for in the generate method and throw an exception if it contains slashes.

Reply:
I think if nothing speaks against using JSON, I'd prefer that. It would be much more future-proof and robust. An exception can only catch literally random cases that happen to contain a slash.

Reply:
Should we generate the JSON directly or rathe pass an array and let the cache river handle how to store it, e.g. as JSON?

Reply:
I've implemented it as array for now